docs/content/development/extensions-core/druid-kerberos.md

```diff
@@ -0,0 +1,70 @@
+|--------|---------------|-----------|-------|--------|
+|`druid.hadoop.security.kerberos.principal`|`druid@EXAMPLE.COM`| Principal user name, used for internal node communication|empty|Yes|
+|`druid.hadoop.security.kerberos.keytab`|`/etc/security/keytabs/druid.headlessUser.keytab`|Path to keytab file used for internal node communication|empty|Yes|
+|`druid.hadoop.security.spnego.principal`|`HTTP/_HOST@EXAMPLE.COM`| SPNego service principal used by druid nodes|empty|Yes|
+|`druid.hadoop.security.spnego.keytab`|`/etc/security/keytabs/spnego.service.keytab`|SPNego service keytab used by druid nodes|empty|Yes|
+|`druid.hadoop.security.spnego.authToLocal`|`RULE:[1:$1@$0](druid@EXAMPLE.COM)s/.*/druid DEFAULT`|It allows you to set a general rule for mapping principal names to local user names. It will be used if there is not an explicit mapping for the principal name that is being translated.|DEFAULT|No|
+|`druid.hadoop.security.spnego.excludedPaths`|`['/status','/health']`| Array of HTTP paths which which does NOT need to be authenticated.|None|No|
+|`druid.hadoop.security.spnego.cookieSignatureSecret`|`secretString`| Secret used to sign authentication cookies|<Random value>|No|
+
+As a note, it is required that the SPNego principal in use by the druid nodes must start with HTTP (This specified by [RFC-4559](https://tools.ietf.org/html/rfc4559)) and must be of the form "HTTP/_HOST@REALM". 
+The special string _HOST will be replaced automatically with the value of config `druid.host`
+
+### Auth to Local Syntax
+
+`druid.hadoop.security.spnego.authToLocal` allows you to set a general rules for mapping principal names to local user names.
+The syntax for mapping rules is `RULE:\[n:string](regexp)s/pattern/replacement/g`. The integer n indicates how many components the target principal should have. If this matches, then a string will be formed from string, substituting the realm of the principal for $0 and the n‘th component of the principal for $n. e.g. if the principal was druid/admin then `\[2:$2$1suffix]` would result in the string `admindruidsuffix`.
+If this string matches regexp, then the s//\[g] substitution command will be run over the string. The optional g will cause the substitution to be global over the string, instead of replacing only the first match in the string.
+If required, multiple rules can be be joined by newline character and specified as a String. 
+
+## Accessing Druid HTTP end points when kerberos security is enabled 
+1. To access druid HTTP endpoints via curl user will need to first login using `kinit` command as follows -  
+
+    ```
+    kinit -k -t <path_to_keytab_file> user@REALM.COM
+    ```
+
+2. Once the login is successful verify that login is successful using `klist` command
+3. Now you can access druid HTTP endpoints using curl command as follows - 
+
+    ```
+    curl --negotiate -u:anyUser -b ~/cookies.txt -c ~/cookies.txt -X POST -H'Content-Type: application/json' <HTTP_END_POINT>
```

Comment:
IMO, user should always specify negotiate. Fwiw, adding this will not mean that authentication handshake will be always done.  
Above curl command works like this - 
1) sends request to that to the server with the cookies for that domain if any.
2) If the server accepts the cookie it will return a 200 OK
3) If the cookie is found to be invalid (multiple reasons - someone restarted the server before the cookie expired, the cookie expired) It will send a response otherwise 401 Unauthorized.
4) The client on receiving 401 now will perform the SPNego negotiate mechanism. 

Will add comment about the use of cookies.